src/solvers/dg_p4est/containers_2d.jl

```diff
@@ -164,17 +172,23 @@ function init_boundaries_iter_face(info, user_data)
   data_array[2] += 1
   mesh = data_array[3]
 
+  # Function barrier because the unpacked user_data above is type-unstable
+  init_boundaries_iter_face_inner(info, boundaries, boundary_id, mesh)
+end
+
+# Function barrier for type stability
+function init_boundaries_iter_face_inner(info, boundaries, boundary_id, mesh)
   # Extract boundary data
-  sides = convert_sc_array(p4est_iter_face_side_t, info.sides)
-  # Global trees array
-  trees = convert_sc_array(p4est_tree_t, mesh.p4est.trees)
-  # Quadrant numbering offset of this quadrant, one-based indexing
-  offset = trees[sides[1].treeid + 1].quadrants_offset
+  side = load_sc_array(p4est_iter_face_side_t, info.sides)
```

Comment:
These are like `unsafe_wrap(Array, ...)` and `unsafe_load`. `convert_sc_array` returns the whole array (as a `Vector`) while `load_sc_array` returns only one entry of the array. In this case (boundaries), there is only one element, so creating an array doesn't make sense. In other cases, I often replaced `convert_sc_array` with a `Tuple` `(load_sc_array(...), load_sc_array(...))` to prevent allocations.

Reply:
OK. The name `load_..._array` sounds to me though as if *this* method actually loads the entire array. Would it make sense to rename it to something more self-descriptive? Or actually create it as an `ScArray{p4est_iter_face_side_t}(info.sides)` which can then be accessed with `[...]` indexing?

Not exactly sure what the best way to go is here, except that from just looking at the code here, I would assume that `side` is an array of type `p4est_iter_face_side_t` using the storage pointed to by `info.sides`. If that's already the correct meaning, maybe we can just leave it as it is ;-)

Reply:
No, `side` is just a `p4est_iter_face_side_t`, no array. How about renaming `convert_sc_array` to `wrap_sc_array` to be consistent with `unsafe_load` and `unsafe_wrap`?

Reply:
Yes, I like `wrap_sc_array` if it returns a similar data type as `unsafe_wrap`. Or maybe `unsafe_wrap_sc`? I'll leave it up to you.

> No, side is just a p4est_iter_face_side_t, no array.

Maybe then something like `sc_get_value` to indicate that this is really returning just a value and not loading the entire array?

Reply:
How about `unsafe_wrap_sc` and `unsafe_load_sc`?

Reply:
Why not? Alternatively: Create an `ScArray{T}` data type and add a new method to base for `unsafe_load` and `unsafe_wrap` that dispatches on `::ScArray`. But maybe that's something for `P4est.jl`, not here. So yes, go ahead with your suggestion!

Reply:
Being able to index sc arrays natively would indeed be a very cool feature of `P4est.jl`.

Reply:
I agree with both (your renaming suggestion and this as a feature of P4est.jl).